### ISSUE TITLE
Support Password Initialization flow

### ISSUE THREAD

**Issue by shimono** [personium/personium-core#341]
# Overview
Support Password Initialization flow by 
introducing a new valid value "initialized"to the "Status"  property of the cell control object "Account".

"initialized" Account's owner is forced to change the credential (password). Then after successful password change, the "Status" value will automatically be changed to "active".

## When Account.Status == initailized

When Account.Status == initailized, the cell should behave like below after a successful authentication.

### ROPC flow
Token endpoint issues special access token with some error code like below.

    {
      "code": "PR412-AN-0000",
      "access_token": "PT_sa....afew",
      "message": "This account is initialized and the password should be changed. ",
      "url": "https://mycell.personium.example/__mypassword"
    }

The issued token can only be used at __mypassword api.  Other endpoints will just treat this token as invalid token. ( returning 401 status code.)
Refresh token is not issued for this special access token that is only used for password change.

### Code/Implicit Flow
Authz Endpoint GUI will prompt password change after successful authentication.  
Then after successful password change, normal access / refresh tokens are issued.

### After password change
Once the password is changed through Authz Endpoint GUI or __mypassword endpoint,
the status value changes to "active".

Probably new token (representing active status) should be issued. 
(TODO HOW?)

## How to initialize password

    POST {cellUrl}__account/me
    {initialize}

    200 OK
    {"tentative-password": "o823jk#448af" }

**Comment by Akira-Ohishi** [comment]
> Support Password Initialization flow by
introducing a new valid value "initialized"to the "Status" property of the cell control object "Account".

For status value, change "initialize" to "passwordChangeRequired".

**Comment by Akira-Ohishi** [comment]
> ### ROPC flow
> Token endpoint issues special access token with some error code like below.

In order to conform to OAuth 2 specification, the response is as follows.
```
{ 
"error":"unauthorized_client",
"error_description":"[PR401-AN-0001] - The password should be changed.",
"access_token":"AP~....(snip)....",
"url":"....(snip)..../__mypassword"
"last_authenticated":1486462510467
"failed_count":2
}
```
* "error" is the error content specified in OAuth 2.
* "error_description" is a combination of message code and message of personium.

**Comment by Akira-Ohishi** [comment]
## How to initialize password

Update Account's status and password.
It does not provide a function to randomly generate the password itself.
An assumption that the application generates and updates.

```
curl "https://cell1.unit1.example/__ctl/Account('account1')"
...(snip)...
-H "X-Personium-Credential:tentative-password"
-d '{"Name":"account1","Status":"passwordChangeRequired"}'
```

* Substitute the account name for password initialization in "account1".
* Substitute the initialized password in "tentative-password".

**Comment by Akira-Ohishi** [comment]
> ### Code/Implicit Flow
> Authz Endpoint GUI will prompt password change after successful authentication.
> Then after successful password change, normal access / refresh tokens are issued.

Along with this, the password change form is added separately from the authentication form.
Since it is necessary to add to Unit setting and Cell property setting, consider it at the time of implementation.

**Comment by SawamiWataru** [comment]
Released v1.7.7
